src/main/java/org/opensearch/timeseries/indices/IndexManagement.java

```diff
@@ -1234,35 +1242,113 @@ protected void rolloverAndDeleteHistoryIndex(
+        RolloverRequest defaultResultIndexRolloverRequest = buildRolloverRequest(resultIndexAlias, rolloverIndexPattern);
+
+        // get config files that have custom result index alias to perform rollover on
+        getConfigsWithCustomResultIndexAlias(ActionListener.wrap(candidateResultAliases -> {
+            if (candidateResultAliases == null || candidateResultAliases.isEmpty()) {
+                // no custom result index alias found
+                if (!doesDefaultResultIndexExist()) {
+                    // no default result index found either
+                    return;
+                }
+                // perform rollover and delete on default result index
+                proceedWithDefaultRolloverAndDelete(
+                    resultIndexAlias,
+                    defaultResultIndexRolloverRequest,
+                    allResultIndicesPattern,
+                    resultIndex
+                );
+                logger.info("Candidate custom result indices are empty.");
+                return;
+            }
+
+            // perform rollover and delete on found custom result index alias
+            candidateResultAliases.forEach(config -> handleCustomResultIndex(config));
+
+        }, e -> {
+            logger.error("Failed to get configs with custom result index alias.", e);
+            // perform rollover and delete on default result index if getting error on getting custom result index alias
+            proceedWithDefaultRolloverAndDelete(resultIndexAlias, defaultResultIndexRolloverRequest, allResultIndicesPattern, resultIndex);
+        }));
+    }
+
+    private void handleCustomResultIndex(Config config) {
+        RolloverRequest rolloverRequest = buildRolloverRequest(
+            config.getCustomResultIndexOrAlias(),
+            getCustomResultIndexPattern(config.getCustomResultIndexOrAlias())
+        );
+
+        // add rollover conditions if found in config
+        if (config.getCustomResultIndexMinAge() != null) {
+            rolloverRequest.addMaxIndexAgeCondition(TimeValue.timeValueDays(config.getCustomResultIndexMinAge()));
+        }
+        if (config.getCustomResultIndexMinSize() != null) {
+            rolloverRequest.addMaxIndexSizeCondition(new ByteSizeValue(config.getCustomResultIndexMinSize(), ByteSizeUnit.MB));
         }
 
-        // We have to pass null for newIndexName in order to get Elastic to increment the index count.
-        RolloverRequest rollOverRequest = new RolloverRequest(resultIndexAlias, null);
+        // perform rollover and delete on AD custom result index alias
+        if (config.getCustomResultIndexOrAlias().startsWith(ADCommonName.CUSTOM_RESULT_INDEX_PREFIX)) {
+            proceedWithRolloverAndDelete(
+                config.getCustomResultIndexOrAlias(),
+                rolloverRequest,
+                getAllCustomResultIndexPattern(config.getCustomResultIndexOrAlias()),
+                (IndexType) ADIndex.CUSTOM_RESULT,
```

Comment:
You can make ADIndex.CUSTOM_RESULT as an input of the calling method and let the sublcass to pass in ADIndex.CUSTOM_RESULT or ForecastIndex.CUSTOM_RESULT.

Reply:
updated this part to not differentiate AD and Forecasting.